File: Add Code Here/C++/String_Palindrome_two_pointer.cpp
#include <iostream>
#include <string>
using namespace std;

bool ispal ( string str )
{
  for( int i = 0, n = str.length(); i < n; ++i )
  {
    if( str[i] != str[n-1-i] )
      return false;
  }
  return true;
}

int main()
{
  string str;
  cout << "Enter a string: ";
  fflush(stdin);
  cin >> str; 
  cout << str << " is " << ( ispal(str) ? "\b" : "not") << " a palindrome." << endl;
  return 0;
}

File: Add Code Here/PYTHON/String_Palindrome_two_pointer.py
def ispal (s):
 for i in range(len(s)):
  if s[i] != s[len(s)-1-i]:
    return False
  return True
s=input("Enter a string value: ")
print(s, "is", "\b" if ispal(s) else "not", "palindrome.")
